src/plugins/intel_gpu/include/intel_gpu/plugin/plugin.hpp

```diff
@@ -18,7 +18,7 @@ namespace intel_gpu {
 class Plugin : public InferenceEngine::IInferencePlugin {
     struct impl;
     std::shared_ptr<impl> _impl;
-    bool isModelCachingEnabled = false;
+    bool isModelCachingEnabled = true;
```

Comment:
Seems like current usage of this flag is incorrect. The flag is stored in the plugin instance, but it depends on the last loaded IR. So I see two issues at least:
1. Not thread safe. Seems like modification of the flag is not guarded with mutex, thus parallel call of compile_model will cause data race
2. Plugin response depends on the models loading sequence:
```cpp
core.compile_model(static_model);
core.compile_model(dynamic_model);
core.get_property("GPU", ov::device::capabilities); // doesn't have IMPORT_EXPORT

core.compile_model(dynamic_model);
core.compile_model(static_model);
core.get_property("GPU", ov::device::capabilities); // have IMPORT_EXPORT
```

If caching for dynamic models can't be enabled quickly, I think we can serialize IR after transformations for dynamic models (similar to CPU plugin behavior) and import will load that IR.

Reply:
Thank you for your review. I'll fix it in another PR.